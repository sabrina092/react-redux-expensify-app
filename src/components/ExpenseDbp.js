import React from 'react' 
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpensesListFilters';
import ExpensesSummary from './ExpensesSummary';
 
export const ExpenseDashboardPage =()=>(
    <div>
      <ExpensesSummary/>
      <ExpenseListFilters/>
      <ExpenseList/>
    </div>
)

export default ExpenseDashboardPage;
