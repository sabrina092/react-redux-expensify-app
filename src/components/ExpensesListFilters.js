import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextfilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters'


export class ExpenseListFilters extends React.Component {
  state = {
    calendarFoucsed: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate)

  };
  onFocusChange = (calendarFoucsed) => {
    this.setState(() => ({ calendarFoucsed }));
  };
  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  };
  onShortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }

  }
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}

        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onShortChange}

        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>

        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          startDateId="your_unique_start_date_id"
          endDate={this.props.filters.endDate}
          endDateId="your_unique_end_date_id"
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFoucsed}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    )
  }
};

const mapStateToProps = (state) => {
  return {
    filters: state.filters
  };
};

const mapDispatchToProps = (dispatch) => ({
  setTextfilter: (text) => dispatch(setTextfilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);

