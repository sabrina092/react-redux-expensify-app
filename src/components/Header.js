import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header =({startLogout})=>(
    <header>
      <h1>Expensify </h1>
      <NavLink to ="/" activeClassName="is-active">Go ExpenseDashboardPage</NavLink>
      <NavLink to ="/create" activeClassName="is-active">Go AddExpense</NavLink>
      <button onClick = { startLogout }> Logout </button>
    </header>
  );
  const mapDispacthToProps =(dispatch)=> ({
     startLogout: ()=> dispatch(startLogout())
  })
  export default connect(undefined, mapDispacthToProps)(Header);