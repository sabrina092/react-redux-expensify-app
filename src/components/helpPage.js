import React from 'react'

const helpPage =() =>(
    <div>
      Help is needed!_!!
    </div>
  )

export default helpPage;