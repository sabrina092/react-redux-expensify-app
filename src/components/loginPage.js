import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';


export const LoginPage = ({ startLogin }) =>(
    <div className= "box-layout">
        
        <div>
            <h1> Expensify Aap </h1>
            <p> It's time to get your expenses under control.</p>
        <button onClick={startLogin}>Login</button>
        </div>
    </div>
);


const mapDispatchToProps = (dispatch) => ({
    startLogin: ()=>  dispatch(startLogin())
});

export default connect (undefined, mapDispatchToProps)(LoginPage)