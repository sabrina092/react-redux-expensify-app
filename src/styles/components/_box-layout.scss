.box-layout {
    background: url('/images/am.jpeg');
    height: 100vh;
    width: 100vw;
}