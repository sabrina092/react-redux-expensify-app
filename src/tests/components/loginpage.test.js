import React from 'react';
import {shallow} from 'enzyme';
import {LoginPage} from '../../components/loginPage'
;


test('should correctly render login page',()=>{
   const wrapper = shallow(<LoginPage/>);
   expect(wrapper).toMatchSnapshot();
   
});

test('should call startlogin on button click',()=>{
   const startLogin = jest.fn();
   const wrapper = shallow(<LoginPage startLogin= {startLogin}/>);
   wrapper.find('button').simulate('click');
   expect(startLogin).toHaveBeenCalled();

});