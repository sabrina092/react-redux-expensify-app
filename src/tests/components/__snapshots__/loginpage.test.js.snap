// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render login page 1`] = `
<div
  className="box-layout"
>
  <div>
    <h1>
       Expensify Aap 
    </h1>
    <p>
       It's time to get your expenses under control.
    </p>
    <button>
      Login
    </button>
  </div>
</div>
`;
